pkg/cmd/server/startconfig/start_allinone.go

```diff
@@ -0,0 +1,136 @@
+following roles:
+
+    $ openshift start master --nodes=<host1,host2,host3,...>
+
+      Launches the server and control plane for OpenShift. You may pass a list of the node
+      hostnames you want to use, or create nodes via the REST API or 'openshift kube'.
+
+You may also pass --etcd=<address> to connect to an external etcd server instead of running an
+integrated instance, or --kubernetes=<addr> and --kubeconfig=<path> to connect to an existing
+Kubernetes cluster.
+`
+
+// NewCommandStartMaster provides a CLI handler for 'start' command
+func NewCommandStartAllInOne() *cobra.Command {
+	options := &AllInOneOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Launch OpenShift All-In-One",
+		Long:  longAllInOneCommandDesc,
+		Run: func(c *cobra.Command, args []string) {
+			if err := options.Validate(); err != nil {
+				glog.Fatal(err)
+			}
+
+			if err := options.StartAllInOne(); err != nil {
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+
+	flags.BoolVar(&options.WriteConfigOnly, "config-only", false, "Indicates that the command should build the config that would be used to start OpenShift and do nothing else.")
+	flags.StringVar(&options.MasterConfigFile, "master-config", "", "Indicates that the command use the config file from a certain location.  If this is specified, all other options are ignored except --config-only.")
+	flags.StringVar(&options.NodeConfigFile, "node-config", "", "Indicates that the command use the config file from a certain location.  If this is specified, all other options are ignored except --config-only.")
+
+	options.MasterArgs = BindMasterArgs(flags)
+	options.NodeArgs = BindNodeArgs(flags)
+
+	bindAddrArg := BindBindAddrArg(flags)
+	options.MasterArgs.BindAddrArg = bindAddrArg
+	options.NodeArgs.BindAddrArg = bindAddrArg
+
+	kubeConnectionArgs := BindKubeConnectionArgs(flags)
+	options.MasterArgs.KubeConnectionArgs = kubeConnectionArgs
+	options.NodeArgs.KubeConnectionArgs = kubeConnectionArgs
+
+	certArgs := BindCertArgs(flags)
+	options.MasterArgs.CertArgs = certArgs
+	options.MasterArgs.KubeConnectionArgs.CertArgs = certArgs
+	options.NodeArgs.CertArgs = certArgs
+
+	imageFormatArgs := BindImageFormatArgs(flags)
+	options.MasterArgs.ImageFormatArgs = imageFormatArgs
+	options.NodeArgs.ImageFormatArgs = imageFormatArgs
+
+	cmd.AddCommand(NewCommandStartMaster())
+	cmd.AddCommand(NewCommandStartNode())
+
+	return cmd
+}
+
+func (o AllInOneOptions) Validate() error {
+	return nil
+}
+
+func (o AllInOneOptions) StartAllInOne() error {
+	glog.Infof("Starting an OpenShift all-in-one, reachable at %s (etcd: %s)", o.MasterArgs.MasterAddr.String(), o.MasterArgs.EtcdAddr.String())
+	glog.Infof("OpenShift master public address is %s", o.MasterArgs.MasterPublicAddr.String())
+
+	masterOptions := MasterOptions{o.MasterArgs, o.WriteConfigOnly, o.MasterConfigFile}
+
+	masterAddr, err := masterOptions.MasterArgs.GetMasterAddress()
```

Comment:
move this first, and output it in the `glog.Infof` call, so we log what we're actually using... maybe the same for etcd and masterpublic addr)